lib/runner.js

```diff
@@ -203,16 +209,21 @@ Runner.prototype.run = function () {
 			}
 		})
 		.then(function () {
-			return self.serial(tests.serial);
+			return tests.only.length ? self.concurrent(tests.only) : [];
+		})
+		.then(function () {
+			return tests.only.length ? [] : self.serial(tests.serial);
 		})
 		.then(function () {
-			return self.concurrent(tests.concurrent);
+			return tests.only.length ? [] : self.concurrent(tests.concurrent);
 		})
 		.then(function () {
 			return eachSeries(tests.after, self._runTest.bind(self));
 		})
 		.catch(noop)
 		.then(function () {
+			// HACK the test count number if only modifier is shown
+			stats.testCount = tests.only.length ? tests.only.length : stats.testCount;
```

Comment:
The variable `testCount` is maintain by the `addTest`, `addOnlyTest`, `addSerialTest` function, which I think that is not suitable. A better way is to return such information (`testCount`, `failCount`) from the promise, then calculate the final stats from at the end of the chain.

However, it involves more code change if doing so. Any ideas?